skyportal/handlers/api/internal/source_views.py

```diff
@@ -0,0 +1,50 @@
+import datetime
+from sqlalchemy import func, desc
+from baselayer.app.access import permissions, auth_or_token
+from ...base import BaseHandler
+from ....models import (
+    DBSession, Source, SourceView, GroupSource
+)
+
+
+class SourceViewsHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        prefs = (self.current_user.preferences if
+                 hasattr(self.current_user, 'preferences') and
+                 self.current_user.preferences else {})
+        if 'topSources' in prefs and 'maxNumSources' in prefs['topSources']:
+            max_num_sources = int(prefs['topSources']['maxNumSources'])
+        else:
+            max_num_sources = 10
+        if 'topSources' in prefs and 'sinceDaysAgo' in prefs['topSources']:
+            since_days_ago = int(prefs['topSources']['sinceDaysAgo'])
+        else:
+            since_days_ago = 30
+        cutoff_day = datetime.datetime.now() - datetime.timedelta(days=since_days_ago)
+        q = (DBSession.query(func.count(SourceView.source_id).label('views'),
```

Comment:
Just for my own education, what is the `label("views")` part?

Reply:
That aggregate count value gets the label "views" here just so that we can refer to that below in `order_by`

Reply:
Just gives that value an alias so you can refer to it later

Reply:
`label("views")` -> `SELECT count(sourceviews.source_id) AS views`, then below there's `ORDER BY views DESC`

Reply:
Nice, that's clever!